src/app/modules/main/wallet_section/collectibles/view.nim

```diff
@@ -9,41 +10,46 @@ QtObject:
   type
     View* = ref object of QObject
       delegate: io_interface.AccessInterface
-      model: Model
-      modelVariant: QVariant
+      model: collections_model.Model
+      flatModel: flat_model.Model
 
   proc delete*(self: View) =
+    self.flatModel.delete
     self.model.delete
-    self.modelVariant.delete
     self.QObject.delete
 
   proc newView*(delegate: io_interface.AccessInterface): View =
     new(result, delete)
     result.QObject.setup
     result.delegate = delegate
     result.model = newModel()
```

Comment:
do we need to keep this one?

Reply:
You mean the old model?
The proxy model doesn't store the data, it just "re-indexes" the data contained in the old model, so we still need this instance. I think it's useful to keep the original model, since it's a logical way to organize the collectibles and we might need it in the future if the design changes.